astropy/io/fits/diff.py

```diff
@@ -459,11 +459,11 @@ def __init__(self, a, b, ignore_keywords=[], ignore_comments=[],
         """
         Parameters
         ----------
-        a : `HDUList`
-            An `HDUList` object.
+        a : `_BaseHDU`
```

Comment:
`_BaseHDU` is private so it is not documented. Please use `BaseHDU` instead, without the backticks,  it will be linked to the appropriate documentation page.
```suggestion
        a : BaseHDU
```
(and same below)

Reply:
Those are now changed. I was surprised that worked but I see `BaseHDU` is manually defined in `conf.py`. Very clever!